tools/walletextension/subscriptions/subscriptions.go

```diff
@@ -0,0 +1,88 @@
+package subscriptions
+
+import (
+	"errors"
+	"fmt"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/log"
+	"github.com/obscuronet/go-obscuro/go/rpc"
+	wecommon "github.com/obscuronet/go-obscuro/tools/walletextension/common"
+	"github.com/obscuronet/go-obscuro/tools/walletextension/userconn"
+)
+
+type SubscriptionManager struct {
+	logger gethlog.Logger
+}
+
+func New(logger gethlog.Logger) *SubscriptionManager {
+	return &SubscriptionManager{
+		logger: logger,
+	}
+}
+
+func (s *SubscriptionManager) HandleNewSubscriptions(clients []rpc.Client, req *wecommon.RPCRequest, resp *interface{}, userConn userconn.UserConn) error {
+	if len(req.Params) == 0 {
+		return fmt.Errorf("could not subscribe as no subscription namespace was provided")
+	}
+
+	if !userConn.SupportsSubscriptions() {
+		return errors.New("userConn doesn't support subscriptions")
+	}
+
+	// create a chanel that will collect the data from all subscriptions
+	commonChannel := make(chan common.IDAndLog)
+
+	// TODO: Create a buffered channel and perform deduplication of logs or implement additional logic to filter logs
+
+	// TODO: Do periodic checks if userConn is closed and unsubscribe from all subscriptions
+
+	// Send all logs from common channel to user (via userConn)
+	go func() {
+		for {
+			select {
+			case idAndLog := <-commonChannel:
+				if userConn.IsClosed() {
+					s.logger.Info("received log but websocket was closed on subscription", log.SubIDKey, idAndLog.SubID)
+					return
+				}
```

Comment:
Should this close the og->obscuro node subscriptions ?

Reply:
I think it should close all the og -> obscuro node subscriptions, since we don't have the user -> OG connection at that moment.
Thanks for pointing that out.  Adding a todo and implementing it 👍